app/presenters/signup_presenter.rb

```diff
@@ -28,6 +28,18 @@ def can_modify_choices?
     choices? && choices_formatted.any?
   end
 
+  def hidden_choices
+    choices.select do |choice|
```

Comment:
This method is calling select but the [return type of the block isn't a boolean](https://apidock.com/ruby/Hash/select) which probably isn't intended.
It's also only used to detect a hidden choice, the hash data we populate here is never used.
I suggest we map the choices into a name and value pair we can use as hidden field attribute values.
ie

```ruby
def hidden_choices
  choices.map do |choice|
    if ignore_facet?(choice["facet_id"])
      choice['facet_choices'].map do |facet_choice|
        {
          name: "filter[#{choice['facet_id']}][]",
          value: facet_choice["key"],
        }
      end
    end
  end.flatten
end
```

This should work with

```ruby
<% hidden_choices.each do |hidden_choice| %>
  <%= hidden_field_tag hidden_choice[:name], hidden_choice[:value] %>
<% end %>
```

as described below.